JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt

```diff
@@ -55,33 +45,77 @@ import com.example.jetnews.ui.navigateTo
 
 @Composable
 fun ArticleScreen(postId: String) {
+    val algebra = +memo { ArticleAlgebra() }
+    val (postState, postStateCb) = +state<ArticleState> { ScreenState.Loading }
 
-    var showDialog by +state { false }
-    // getting the post from our list of posts by Id
-    val post = _posts.find { it.id == postId } ?: return
+    fun load(): Disposable = algebra.getArticle(postId, postStateCb).unsafeRunAsyncCancellable { }
+
+    +onActive {
+        onDispose(load())
+    }
+
+    PostStateLCE(postState) {
+        //TODO how to dispose this?
+        load()
+    }
+}
 
+@Composable
+fun PostStateLCE(articleState: ArticleState, retryClick: () -> Unit) {
+    var showDialog by +state { false }
     if (showDialog) {
         FunctionalityNotAvailablePopup {
             showDialog = false
         }
     }
 
     Column {
+        var topAppBarTitle by +state { "" }
```

Comment:
I'd rather suggest to declare it as val and use destructuring to avoid local mutability, same above with `showDialog`